### PR TITLE
HDDS-5609. Document CI workflow

### PR DESCRIPTION
**Pull request by GeorgeJahad**
## What changes were proposed in this pull request?

Document CI workflow

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5609

## How was this patch tested?

No source code changes

### CODE REVIEW

**Comment by GeorgeJahad** [comment]
@fapifta This is my attempt at the doc you requested.  Would you mind reviewing?

File: .github/ci.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+#### coverage job
+[Merges](../hadoop-ozone/dev-support/checks/coverage.sh) the coverage data from the following jobs that were run earlier:
+- acceptance
+- basic
+- integration
+
+### Cancelling Workflow
+[This](./workflows/cancel-ci.yaml) workflow is triggered each time a [build-branch](ci.md#build-branch-workflow) workflow is triggered.  It monitors all the jobs triggered by that workflow run for failure and cancels any continuing jobs after one fails.  This reduces our GA usage.
+
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply .github/ci.md:66]: The `fail-fast` flag only applies within parallel jobs triggered by matrix config (compile, basic, acceptance and integration).  E.g. if _acceptance (secure)_ fails, then other _acceptance (*)_ are cancelled by GitHub due to this flag, but _integration (*)_ are unaffected.  The `cancel-ci` workflow works across jobs.

**Review by adoroszlai — COMMENTED** [review]
Thanks @GeorgeJahad for writing this doc.  Mostly looks good.  I have provided few clarifications.

File: .github/ci.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+[Runs](../hadoop-ozone/dev-support/checks/acceptance.sh) smoketests using robot framework and a real docker compose cluster
+
+#### kubernetes job
+[Runs](../hadoop-ozone/dev-support/checks/kubernetes.sh) k8s tests
+
+#### integration job
+[Runs](../hadoop-ozone/dev-support/checks/integration.sh) 'mvn test' for all integration/minicluster tests, split into 3 subjobs:
+- client
+- filesystem-hdds
+- ozone
```
[the remark this answers is not in the document]
Comment (GeorgeJahad) [reply .github/ci.md:56]: Just the integration subjobs or all the subjobs?

File: .github/ci.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+- [Build](https://github.com/apache/ozone/actions/workflows/main.yml)
+- [build-branch](https://github.com/apache/ozone/actions/workflows/chaos.yml)
+- [pr-check](https://github.com/apache/ozone/actions/workflows/pr.yml)
+
+Note that the deprecated [build-branch](https://github.com/apache/ozone/actions/workflows/chaos.yml) has the same name as the current [build-branch](https://github.com/apache/ozone/actions/workflows/post-commit.yml).  (They can be distinguished by the URL.)
+
+
+## Tips
+
+- When a build of the Ozone master branch fails, it's artifacts are stored [here](https://elek.github.io/ozone-build-results/).
```
[the remark this answers is not in the document]
Comment (GeorgeJahad) [reply .github/ci.md]: Thanks @adoroszlai! (I'm embarassed to see I still get the possessive mixed up with contractions.)  Good thing I have you to check my English!  (Can you help me with my Magyar too?)

File: .github/ci.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adoroszlai) [reply .github/ci.md]: Only these integration ones.

**Review by adoroszlai — APPROVED** [review]
Thanks @GeorgeJahad for updating the doc, LGTM.

**Comment by errose28** [comment]
Thanks for writing this @GeorgeJahad, it was a very interesting read!

**Comment by GeorgeJahad** [comment]
> it was a very interesting read!
Thanks to @adoroszlai for doing all the hard work.

**Comment by GeorgeJahad** [comment]
FYI: @swagle asked me to add a wiki page: https://cwiki.apache.org/confluence/display/OZONE/Ozone+CI+with+Github+Actions

**Comment by adoroszlai** [comment]
> @fapifta This is my attempt at the doc you requested. Would you mind reviewing?

@fapifta Please let us know if you have any comments, or if I can merge this.  Thanks.

**Comment by adoroszlai** [comment]
Thanks @GeorgeJahad for the contribution, @errose28 for the review.

**Pull request merged**